data-protocols/ids/ids-identity-provider/src/main/java/org/eclipse/dataspaceconnector/ids/daps/validation/ValidationRuleResult.java

```diff
@@ -0,0 +1,58 @@
+/*
+ *  Copyright (c) 2020, 2021 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.ids.daps.validation;
+
+import java.util.Objects;
+
+/**
+ * Evaluation result of a custom rule, can be either success or failure.
+ */
+public class ValidationRuleResult {
+
+    private final boolean success;
+    private final String message;
```

Comment:
It might make sense to actually have a `List<String>` here, since there might be multiple violations, since we can have multiple rules, and each rule might validate multiple things.